File: Models/Movement.cs
namespace movements.Models;

public class Movement : ITimestamp
{
    public Guid ID { set; get; }
    public string Name { set; get; } 
    public string Description { set; get; }
    public int Amount { get; set; }
    public DateTime Date { get; set; }
    public MovementType Type { get; set; }
    public string Status { get; set; }
    public Guid UserID { get; set; }
    public Guid? PermanentCorrelationID { set; get; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: ClientApp/src/app/movements/movement.ts
export enum MovementType {
  expense = 0,
  income = 1,
}

export enum MovementStatus {
  pending = 'pending',
  done = 'done',
  deleted = 'deleted',
}

export type Movement = {
  id: string
  name: string
  amount: number
  description: string
  date: Date
  type: MovementType
  status: MovementStatus
  permanentCorrelationID: string | null
  virtual: boolean
  userID: string
  end: Date | null
}
